protocol/error.go

```diff
@@ -0,0 +1,32 @@
+// Defines constants representing the types
+// of errors that the server may return to a client.
+
+package protocol
+
+import "errors"
+
+type ErrorCode int
+
+const (
+	Success ErrorCode = iota + 10
+	ErrorInternalServer
+	ErrorNameExisted
+	ErrorNameNotFound
+	ErrorMalformedClientMessage
+)
+
+var (
+	errorMessages = map[ErrorCode]error{
+		ErrorMalformedClientMessage: errors.New("[coniks] Malformed client request"),
+		ErrorNameExisted:            errors.New("[coniks] Registering identity is already registered"),
+		ErrorNameNotFound:           errors.New("[coniks] Name not found"),
+		ErrorInternalServer:         errors.New("[coniks] Internal server error"),
```

Comment:
Yeah, I agree. Also because I think it's best to keep the server and protocol logic separate.